### PR TITLE
TT-5947 Refactor/mongo client interface

### PR DESCRIPTION
**Pull request by tbuchaillot**
## Description

This PR creates a mgo interface under `internal/mgo`. This interface call mgo function under the hood.
It also modifies mongo pump code to utilize this interface - there are not a lot of changes.

All the mgo interface mocks are created using [mockery](https://github.com/mockery/mockery) using `mockery --all `command and all the existing tests were modified to use mocks + added new scenarios in each one, increasing test coverage of `CollectionExists`, `CapCollection`, and `EnsureIndexes` functions to almost 100%. 
This PR also adds a basic test for `WriteData` function.

## Motivation and Context

https://tyktech.atlassian.net/browse/TT-5947
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -19,12 +19,14 @@ require (
 	github.com/go-ole/go-ole v1.2.4 // indirect
 	github.com/go-redis/redis/v8 v8.3.1
 	github.com/gocraft/health v0.0.0-20170925182251-8675af27fef0
+	github.com/golang/mock v1.3.1
```
Comment (sredxny) [inline go.mod:22]: Where are we using it? I don't see a place where its imported 🤔

File: pumps/internal/mgo/database.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package mgo
+
+import mgo "gopkg.in/mgo.v2"
+
+type DatabaseManager interface {
+	// AddUser(username, password string, readOnly bool) error
```
Comment (sredxny) [inline pumps/internal/mgo/database.go:6]: please remove if it will not be used

File: go.mod (hunk 1 above)
Comment (sredxny) [reply go.mod:22]: Instead I can see that we import `github.com/stretchr/testify/mock`

**Review by sredxny — CHANGES_REQUESTED** [review]

**Review by mativm02 — APPROVED** [review]
lgtm

File: go.mod (hunk 1 above)
Comment (tbuchaillot) [reply go.mod]: removed @sredxny. `github.com/stretchr/testify/mock ` is included in `github.com/stretchr/testify `

File: pumps/internal/mgo/database.go (hunk 2 above)
Comment (tbuchaillot) [reply pumps/internal/mgo/database.go]: removed

**Review by sredxny — APPROVED** [review]
LGTM. Please check the test pipeline that is failing

**Pull request merged**